<a href="https://colab.research.google.com/github/MohdFuzailHaider/flyrank/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I'm choosing lane 2 ie **Refresh / Content Opportunity Scoring.**

---


I choose this because i think that is addresses a practical issue that most bussiness face which is prioritizing content for review.\
Since the SEO team have mutliple pages to manage if they will get a ranking on what pages to review first based on refresh, monitoring, or improvement based on observable search and engagement signals.\
Bigger issues will get fixed before the less priority issues.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision**\
My work aims to improve the decision of which pages should be reviewed first.\
**Action**\
The SEO team uses the ranked recommendation to decide which pages to refresh, expand, monitor,protect or prune.\
**Cost of a wrong recommendation**\
If a page is incorrectly prioritized, the team may spend time reviewing content that doesn't need attention. If a important page is missed, valuable traffic or engagement opportunites maybe lost.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
import pandas as  pd
df = pd.read_csv("/content/content_refresh_anonymized.csv")
total_pages= len(df)
declining_pages = (df["trend_direction"] == "down").sum()
declining_pct = (declining_pages / total_pages)* 100
avg_impression = df["impressions_90d"].mean()
print(f"Total pages: {total_pages}")
print(f"Declining pages: {declining_pages}")
print(f"Percentage declining: {declining_pct:.2f}%")
print(f"Average impressions (90d): {avg_impression:.2f}")

Total pages: 30000
Declining pages: 16262
Percentage declining: 54.21%
Average impressions (90d): 5200.37


1. The starter dataset contains 30,000 content pages, providing a sufficiently large sample for analyzing content performance.
2. Among these pages, 16,262 (54.21%) are labeled as declining, indicating that more than half of the content shows signs of downward performance under the starter definition.
3. Additionally, the average number of impressions over the last 90 days is 5,200.37, suggesting that many pages have meaningful search visibility.
4. These statistics indicate that the dataset contains enough variation in content performance to build and evaluate a data-driven prioritization system for content review.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project uses observable search and engagement signals to support decisions about which pages may deserve review.\
The results will identify patterns and proved a ranked list of pages based on measured data, but they don't prove that refressing page will imporve its performance.\
This analysis is observational and inteded for decision support rather than causal inderence.\
Likewaise, the project doesn't predict Google's ranking algorithm or gaurantee that any recommended action will increase traffic.\
Final decisions should always involve human review and business context.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.